### PR TITLE
Feature: Use "Distribution" certificate for iOS App Store and Mac App Store profiles

### PR DESCRIPTION
**Pull request by priitlatt**
**Background information**

Xcode 11 introduced a new kind of signing certificate _Apple Distribution_ with type `DISTRIBUTION`. It allowed using a single certificate to code sign all types of applications (iOS, tvOS, MacOS, etc), as opposed to the previous approach which required dedicated distribution certificate for each target platform. Until now provisioning profiles of type `IOS_APP_STORE` and `MAC_APP_STORE` were assigned a certificate with type `IOS_DISTRIBUTION` and `MAC_APP_DISTRIBUTION`, respectively, when downloaded (or created) using `app-store-connect fetch-signing-files`. This prevented reusing one signing certificate for signing apps that target different platforms, and created excess certificates in Apple Developer Portal.

Changes here update default provisioning profile type to certificate type mapping in the following way:
- `IOS_APP_STORE` provisioning profiles use `DISTRIBUTION` certificates by default (previously `IOS_DISTRIBUTION`),
- `MAC_APP_STORE` provisioning profiles use `DISTRIBUTION` certificates by default (previously `MAC_APP_DISTRIBUTION`).
Note that the change is **backwards compatible** in the sense that existing matching `IOS_DISTRIBUTION` and `MAC_APP_DISTRIBUTION` certificates are still used for respective provisioning profiles in case they are found and no `DISTRIBUTION` certificates are present.

Induced changes include possibility to list certificates using `app-store-connect list-certificates` by specifying more than certificate type by `--type` option.

There is also **one deprecation** in the Python API. First argument of `AppStoreConnect.list_certificates` was changed from
```python
certificate_type: Optional[CertificateType] = None
```
to
```python
certificate_types: Optional[Union[CertificateType, Sequence[CertificateType]]] = None
```
The **change is fully backwards compatible** for now. Positional usage works exactly as it used to, and a warnign is logged when the deprecated keyword argument is used.

**Updated actions:**
- `app-store-connect fetch-signing-files`,
- `app-store-connect list-certificates`.

### CODE REVIEW

**Comment by priitlatt** [comment]
Follow up to #165.

**Comment by priitlatt** [comment]
Keeping @dan-os in the loop, who originally pointed this out.

**Comment by priitlatt** [comment]
Fixes #158.

**Review by artemii-yanushevskyi — APPROVED** [review]

File: src/codemagic/apple/resources/enums.py [hunk 1]
```diff
@@ -190,13 +189,13 @@ class CertificateType(ResourceEnum):
...
         if profile_type is profile_type.IOS_APP_ADHOC:
             return CertificateType.IOS_DISTRIBUTION
         elif profile_type is profile_type.IOS_APP_DEVELOPMENT:
             return CertificateType.IOS_DEVELOPMENT
         elif profile_type is profile_type.IOS_APP_STORE:
-            return CertificateType.IOS_DISTRIBUTION
+            return CertificateType.DISTRIBUTION
         elif profile_type is profile_type.MAC_APP_DEVELOPMENT:
             return CertificateType.MAC_APP_DEVELOPMENT
         elif profile_type is profile_type.MAC_APP_STORE:
```
Comment (artemii-yanushevskyi) [inline src/codemagic/apple/resources/enums.py:201]: macOS is still using `MAC_APP_DISTRIBUTION` by default. It's better to use `DISTRIBUTION` by default instead

**Review by artemii-yanushevskyi — CHANGES_REQUESTED** [review]

File: src/codemagic/apple/resources/enums.py (hunk 1 above)
Comment (priitlatt) [reply src/codemagic/apple/resources/enums.py:201]: Resolved in c2686c0. PR description and changelog have also recieved relevant updates.

**Review by artemii-yanushevskyi — APPROVED** [review]

**Pull request merged**